src/utils.py

```diff
@@ -1,109 +1,301 @@
-    scale_factor = pixel_size / target_pixel_size
-    img = (img / scale_factor).astype(np.uint8)
-    return img
-
-def load_bids_image(image_path, pixel_size_metadata=None):
-    """Loads a BIDS-formatted image and adjusts pixel values if needed."""
-    img = cv2.imread(image_path)
-    if pixel_size_metadata:
-        img = adjust_pixel_values(img, pixel_size_metadata)
-    return img
-
-def resize_and_pad(img, target_size=(416, 416)):
-    """Resizes an image while maintaining aspect ratio and pads to target size."""
-    old_size = img.shape[:2]
-    ratio = min(target_size[0] / old_size[0], target_size[1] / old_size[1])
-    new_size = tuple([int(x * ratio) for x in old_size])
-
-    img = cv2.resize(img, (new_size[1], new_size[0]))
-
-    delta_w = target_size[1] - new_size[1]
-    delta_h = target_size[0] - new_size[0]
-    top, bottom = delta_h//2, delta_h-(delta_h//2)
-    left, right = delta_w//2, delta_w-(delta_w//2)
-
-    img = cv2.copyMakeBorder(img, top, bottom, left, right, cv2.BORDER_CONSTANT, value=[0, 0, 0])
-    return img
-
-def normalize_and_window(img):
-    """Normalizes and applies intensity windowing to an image."""
-    img = (img - img.min()) / (img.max() - img.min())  # Normalize to [0, 1]
-    low_percentile, high_percentile = np.percentile(img, (2, 98))
-    img = np.clip(img, low_percentile, high_percentile)
-    img = (img - img.min()) / (img.max() - img.min())  # Renormalize
-    img = (img * 255).astype(np.uint8)  # Scale to [0, 255]
-    return img
-
-def find_regions(img):
-    """Finds connected regions directly from a binary segmentation mask."""
-    # Ensure the image is of integer type for labeling
-    img = img.astype(int)
-
-    # Label connected regions
-    labeled_img = measure.label(img)
-
-    # Extract region properties
-    regions = measure.regionprops(labeled_img)
-    if len(regions) == 0:
-        print("No regions found!")
-
-    return regions
-
-def crop_and_resize(img, region, target_size=(416, 416)):
-    """Crops an image around a region and resizes it."""
-    if region.area == 0:
-        print("Region area is zero!")
-        return np.zeros((target_size[0], target_size[1], 3), dtype=np.uint8)  # Return a black image
-
-    minr, minc, maxr, maxc = region.bbox
-    cropped_img = img[minr:maxr, minc:maxc]
-    return resize_and_pad(cropped_img, target_size)
+from tqdm import tqdm
+
+from utils import data_split
+
+SEM_DATASET_URL = "https://github.com/axondeepseg/data_axondeepseg_sem"
+
+# Reorganizes data structure into COCO (retinanet)
+
+# Other problem: Bounding box surrounding myelin and axon (only axon)
+
+
+def save_yolo_dset(image_mask_pairs, image_dir, mask_dir):
+    for image_name, img, mask_path in image_mask_pairs:
+        # Save the image
+        cv2.imwrite(os.path.join(image_dir, image_name), img)
+        
+        # Copy the corresponding label (mask) file
+        shutil.copy(mask_path, os.path.join(mask_dir, os.path.basename(mask_path)))

```

Comment:
The Path library (which was removed from the imports list) makes it easier to handle mixing directory paths and filenames, and is more OS agnostic (ie would work on Windows). Not a requirement, but it’s good practice. See the AxonDeepSeg files on examples of how this library is used in different situations

Reply:
Eg in this test file’s setup steps: https://github.com/axondeepseg/axondeepseg/blob/7b79f486f731dfa05f9fbfab1a24e4710a652c9b/test/morphometrics/test_compute_morphometrics.py#L46

Reply:
will be fixed in another PR